test/witan/send/adroddiad/tablecloth_utils_test.clj

```diff
@@ -0,0 +1,582 @@
+                              {:a 1, :b 1, :c 2}
+                              {:a 1, :b 2, :c 1}
+                              {:a 2, :b 1, :c 1}])]
+          (= (-> ds tc-utils/select-non-unique-rows)
+             (-> ds tc-utils/non-unique-by)))))
+
+  (testing "Replaces deprecated `select-non-unique-key-rows` function."
+    (is (let [ds (tc/dataset [{:a 1, :b 1, :c 1}
+                              {:a 1, :b 1, :c 1}
+                              {:a 1, :b 1, :c 2}
+                              {:a 1, :b 2, :c 1}
+                              {:a 2, :b 1, :c 1}])]
+          (= (-> ds (tc-utils/select-non-unique-key-rows [:a :b]))
+             (-> ds (tc-utils/non-unique-by [:a :b])))))))
+
+
+(deftest non-unique-by-keys
+  (testing (str "With no arguments returns a dataset containing one example of "
+                "each row that appears multiple times in the dataset.")
+    (is (=
+         (-> (tc/dataset [{:a 1, :b 1, :c 1}
+                          {:a 1, :b 1, :c 1}
+                          {:a 1, :b 1, :c 2}
+                          {:a 1, :b 2, :c 1}
+                          {:a 2, :b 1, :c 1}])
+             tc-utils/non-unique-by-keys)
+         (tc/dataset [{:a 1, :b 1, :c 1}]))))
+
+  (testing "Similarly with `columns-selector` `:all`."
+    (is (=
+         (-> (tc/dataset [{:a 1, :b 1, :c 1}
+                          {:a 1, :b 1, :c 1}
+                          {:a 1, :b 1, :c 2}
+                          {:a 1, :b 2, :c 1}
+                          {:a 2, :b 1, :c 1}])
+             (tc-utils/non-unique-by-keys :all))
+         (tc/dataset [{:a 1, :b 1, :c 1}]))))
+
+  (testing (str "Which is essentially the same as"
+                "`tc-utils/non-unique-by` followed by `tc/unique-by` "
+                "(but simpler).")
+    (is (let [ds (tc/dataset [{:a 1, :b 1, :c 1}
+                              {:a 1, :b 1, :c 1}
+                              {:a 1, :b 1, :c 2}
+                              {:a 1, :b 2, :c 1}
+                              {:a 2, :b 1, :c 1}])]
+          (= (-> ds tc-utils/non-unique-by-keys)
+             (-> ds tc-utils/non-unique-by tc/unique-by)))))
+
+  (testing (str "With a subset of columns specified, "
+                "returns a dataset with the combinations of these columns that "
+                "appear more than once in the dataset: I.e. the non-unique keys.")
+    (is (=
+         (-> (tc/dataset [{:a 1, :b 1, :c 1}
+                          {:a 1, :b 1, :c 1}
+                          {:a 1, :b 1, :c 2}
+                          {:a 1, :b 2, :c 1}
+                          {:a 2, :b 1, :c 1}])
+             (tc-utils/non-unique-by-keys [:a :b]))
+         (tc/dataset [{:a 1, :b 1}]))))
+
+  (testing (str "Which is essentially the same as"
+                "`tc-utils/non-unique-by` "
+                "followed by `tc/select-columns` for the key columns "
+                "followed by `tc/unique-by` "
+                "(but simpler).")
```

Comment:
simpler or easy (given that these are important words in clojure)

Reply:
I suppose easier, assuming simpler refers to the computers perspective? 
Have removed to avoid.